### PR TITLE
Remove the OAuthClients traversal

### PR DESCRIPTION
**Pull request by jon-betts**
Previously we had:

 * A traversal which looked up the OAuth client
 * Which returned the model as a context (something we've been trying to prevent)
 * ... or raised `KeyError` which results in a 404 when raised during traversal

Now:

* The view looks up the client
* And raises `HTTPNotFound` if it can't get one

This was only used in one place and unnecessarily complicated how this works.

### CODE REVIEW

File: h/views/admin/oauthclients.py [hunk 1]
```diff
@@ -132,6 +133,11 @@ def on_success(appstruct):
             on_failure=self._template_context,
         )
 
+    @view_config(request_method="POST", request_param="delete")
+    def delete(self):
+        self.request.db.delete(self.client)
+        return HTTPFound(location=self.request.route_url("admin.oauthclients"))
+
```
Comment (jon-betts) [inline h/views/admin/oauthclients.py:140]: Just moving this above the private functions

File: h/views/admin/oauthclients.py [hunk 2]
```diff
@@ -150,7 +156,12 @@ def _update_appstruct(self):
...
-        return HTTPFound(location=self.request.route_url("admin.oauthclients"))
+    @classmethod
+    def _get_client(cls, request):
+        client_id = request.matchdict.get("id")
+
+        try:
+            return request.db.query(AuthClient).filter_by(id=client_id).one()
+        except (NoResultFound, StatementError) as err:
+            # Statement errors happen if the id is invalid
+            raise HTTPNotFound() from err
```
Comment (jon-betts) [inline h/views/admin/oauthclients.py:167]: I don't think having this in a separate file makes anything easier to test or understand

File: tests/h/views/admin/oauthclients_test.py [hunk 3]
```diff
@@ -136,92 +137,116 @@ def test_post_redirects_to_edit_view(self, form_post, matchers, pyramid_request)
...
+
+        with pytest.raises(HTTPNotFound):
+            AuthClientEditController(sentinel.context, pyramid_request)
+
+    def test_read(self, pyramid_request, auth_client):
+        controller = AuthClientEditController(sentinel.context, pyramid_request)
+
+        response = controller.read()
+
+        assert response["form"] == self._expected_form(auth_client)
```
Comment (jon-betts) [inline tests/h/views/admin/oauthclients_test.py:164]: This is the new testing to cover the get method

File: h/traversal/auth_client.py [hunk 4]
```diff
@@ -1,36 +0,0 @@
...
-from h.models import AuthClient
-from h.traversal.root import Root, RootFactory
-
-
-class AuthClientRoot(RootFactory):
-    """
-    Root factory for routes whose context is an :py:class:`h.traversal.AuthClientContext`.
-
-    FIXME: This class should return AuthClientContext objects, not AuthClient
-    objects.
```
Comment (marcospri) [inline h/traversal/auth_client.py:13]: Fixed!

**Review by marcospri — APPROVED** [review]
Nice, I'm sure there's a good use case for the traversal solution but I don't think this was it :+1:

**Pull request merged**